mesonbuild/dependencies/pkgconfig.py

```diff
@@ -157,15 +262,57 @@ def setup_env(env: EnvironOrDict, environment: 'Environment', for_machine: Machi
     def _call_pkgbin(self, args: T.List[str], env: T.Optional[EnvironOrDict] = None) -> T.Tuple[int, str, str]:
         assert isinstance(self.pkgbin, ExternalProgram)
         env = env or os.environ
-        env = PkgConfigDependency.setup_env(env, self.env, self.for_machine)
+        env = self.setup_env(env, self.env, self.for_machine)
 
         fenv = frozenset(env.items())
         targs = tuple(args)
-        cache = PkgConfigDependency.pkgbin_cache
+        cache = self.pkgbin_cache
         if (self.pkgbin, targs, fenv) not in cache:
             cache[(self.pkgbin, targs, fenv)] = self._call_pkgbin_real(args, env)
         return cache[(self.pkgbin, targs, fenv)]
 
+
+class PkgConfigDependency(ExternalDependency):
+
+    def __init__(self, name: str, environment: Environment, kwargs: T.Dict[str, T.Any], language: T.Optional[str] = None) -> None:
+        super().__init__(DependencyTypeName('pkgconfig'), environment, kwargs, language=language)
```

Comment:
@eli-schwartz I wonder if we should change this to pkg-config and deprecate pkgconfig similar to your other PR

Reply:
We probably should, but I'll leave that to another PR that deals with that specifically.